### PR TITLE
Network enrichment script

### LINKED ISSUE #46: Network data enrichment
Enrich data in network alerts using virustotal

### PR DESCRIPTION
**Pull request by karpp**
Closes #46

### CODE REVIEW

**Comment by karpp** [comment]
Deployment script is not ready

**Review by niedziol — CHANGES_REQUESTED** [review]

File: capstone/network-enrichment/custom_vt_network.py [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+    sock = socket(AF_UNIX, SOCK_DGRAM)
+    sock.connect(socket_addr)
+    sock.send(string.encode())
+    sock.close()
+
+def check_existing_ip(ip):
+    req = {'query': {'bool': {'must': [{'match': {'data.integration': 'custom-vt-network'}}, {'match': {'data.srcip': ip}}]}}}
+    resp = requests.get('http://elastic-stack-instance:9200/wazuh-alerts-3.x-*/_search', json=req)
+    if resp.status_code != 200:
+        logging.error('Request to elasticsearch failed')
```
Comment (mgmacias95) [inline capstone/network-enrichment/custom_vt_network.py:36]: Add response text from elastic search

File: capstone/network-enrichment/custom_vt_network.py [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+    alert_file = open(alert_filename)
+
+    alert_json = json.loads(alert_file.read())
+    alert_file.close()
+
+    if not check_existing_ip(alert_json['data']['srcip']):
+        alert_output = request_virustotal_info(alert_json, api_key)
+        send_event(alert_output, alert_json['agent'])
+    else:
+        logging.info('IP found again: ' + alert_json['data']['srcip'])
```
Comment (mgmacias95) [inline capstone/network-enrichment/custom_vt_network.py:93]: f-string

File: capstone/network-enrichment/custom_vt_network.py [hunk 3]
```diff
@@ -0,0 +1,106 @@
+#!/usr/bin/env python3
+  
+import json
+import logging
+import os
+import requests
+import sys
+import time
```
Comment (mgmacias95) [inline capstone/network-enrichment/custom_vt_network.py:8]: is this used?

**Review by mgmacias95 — CHANGES_REQUESTED** [review]

File: capstone/network-enrichment/deploy.sh [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+scp custom_vt_network.py wazuh-manager:custom-vt-network
+ssh wazuh-manager << EOF
+    sudo su
+    mv custom-vt-network /var/ossec/integrations/custom-vt-network
+    chmod 750 /var/ossec/integrations/custom-vt-network
+    chown root:ossec /var/ossec/integrations/custom-vt-network
+
+    OSSECCONF=$(cat /var/ossec/etc/ossec.conf)
+    if [[ "$OSSECCONF" != *"custom-vt-network"* ]]; then
```
Comment (niedziol) [inline capstone/network-enrichment/deploy.sh:11]: This is not working, you have multiple copy of the same config in `ossec.conf`.

**Review by niedziol — CHANGES_REQUESTED** [review]

**Review by niedziol — APPROVED** [review]
LGTM

**Pull request merged**